sdk/provisioning/Azure.Provisioning/src/Resource.cs

```diff
@@ -191,6 +193,16 @@ private void ValidateOverrideCanBeAdded(object instance, string name)
             }
         }
 
+        /// <summary>
+        /// Assigns a role to the resource.
+        /// </summary>
+        /// <param name="roleDefinition">The role definition.</param>
+        /// <param name="principalId">The principal ID.</param>
+        public RoleAssignment AssignRole(RoleDefinition roleDefinition, Guid? principalId = default)
```

Comment:
we don't want this to depend on Azure.Provisioning.Authorization so add this as an extension method from that namespace